src/node_port.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling mongos Kubernetes resources for a single mongos pod."""
+
+import logging
+from functools import cached_property
+from ops.charm import CharmBase
+from lightkube.models.meta_v1 import ObjectMeta
+from lightkube.core.client import Client
+from lightkube.core.exceptions import ApiError
+from lightkube.resources.core_v1 import Pod, Service
+from lightkube.models.core_v1 import ServicePort, ServiceSpec
+
+
+logger = logging.getLogger(__name__)
+
+# default logging from lightkube httpx requests is very noisy
+logging.getLogger("lightkube").disabled = True
+logging.getLogger("lightkube.core.client").disabled = True
+logging.getLogger("httpx").disabled = True
+logging.getLogger("httpcore").disabled = True
+
+
+class NodePortManager:
+    """Manager for handling mongos Kubernetes resources for a single mongos pod."""
+
+    def __init__(
+        self,
+        charm: CharmBase,
+        pod_name: str,
+        namespace: str,
+    ):
+        self.charm = charm
+        self.pod_name = pod_name
+        self.app_name = "-".join(pod_name.split("-")[:-1])
+        self.namespace = namespace
```

Comment:
**suggestion:** In our 'glorious' restructuring of the charm, we don't pass `charm` everywhere. That's why we infer everything from the pod-name, as it's not dependent on any `ops`-y stuff.

I think for you, it might just be easier to remove these from the object constructor, and set all these attrs to:

```python
self.pod_name = self.charm.unit.name.replace("/", "-")
self.app_name = self.charm.app.name
self.namespace = self.charm.model.name
```